### ISSUE TITLE
[Bug]: Unable to configure features through URL parameters

### ISSUE THREAD

**Issue by limhenry** [storybookjs/storybook#23550]
### Describe the bug

Unable to configure features (full screen, nav, panel, etc ...) through URL parameters: https://storybook.js.org/docs/react/configure/features-and-behavior#configuring-through-url-parameters

### To Reproduce

Trying to set panel position through URL parameters (I am using Yann's repo as example):

V7.1 (Not working):

- Right Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=right
- Bottom Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=bottom
- No Panel: https://mealdrop.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=false

V6.4.7 (Working):

- Right Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=right
- Bottom Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=bottom
- No Panel: https://mealdrop-git-feat-fix-test-ci-yannbf.vercel.app/storybook/?path=/story/components-button--icon-and-text&panel=false

_(Try to open link in Incognito mode for "best" result)_

---

The same issue applies to [other features/query params](https://storybook.js.org/docs/react/configure/features-and-behavior#configuring-through-url-parameters) (`shortcuts`, `full`, `nav`, `panel`, etc ...)

### System

```shell
Environment Info:

  System:
    OS: macOS 13.4.1
    CPU: (12) arm64 Apple M2 Max
  Binaries:
    Node: 16.19.1 - ~/.nvm/versions/node/v16.19.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.1/bin/yarn
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.1/bin/npm
  Browsers:
    Safari: 16.5.2
  npmPackages:
    @storybook/addon-a11y: ^7.1.0 => 6.5.13 
    @storybook/addon-actions: ^7.1.0 => 6.5.13 
    @storybook/addon-essentials: ^7.1.0 => 6.5.13 
    @storybook/addon-interactions: ^7.1.0 => 6.5.13 
    @storybook/addon-links: ^7.1.0 => 6.5.13 
    @storybook/mdx2-csf: ^1.1.0 => 0.0.3 
    @storybook/preset-scss: ^1.0.3 => 1.0.3 
    @storybook/testing-library: 0.2.0 => 0.0.13 
    @storybook/vue3: ^7.1.0 => 6.5.13
```

### Additional context

We are embedding Storybook in other website, and with lacking options to hide side panel through URL parameters is causing us to show unrelated/irrelevant information on the website. And not able to change the panel position is also causing layout issue to us.

**Comment by limhenry** [comment]
So I decided to spend some time _starring_ at the code, and I actually found the reason why URL parameters is broken:

The `init: () => ...` in [layout.ts](https://github.com/storybookjs/storybook/blob/next/code/lib/manager-api/src/modules/layout.ts#L285) added in #18550 is somehow overriding the URL parameters values from [url.ts](https://github.com/storybookjs/storybook/blob/next/code/lib/manager-api/src/modules/url.ts).

By removing that `init: () => ...`, URL parameters is now working fine again.

However, I guess it will break the "prebuild manager"? (although I have no idea what is this 😅)

---

Related Commit: https://github.com/storybookjs/storybook/pull/18550/commits/bd8a2204fb3efd55a7fc13f1c27ec3ec3658ddf9

cc @ndelangen @yannbf

**Comment by TimKolberger** [comment]
This bug affects the integration of InVision with Storybook: https://support.invisionapp.com/docs/embedding-storybook-in-dsm-via-url

The sidebar is visible in all storybook iframes.

**Comment by yannbf** [comment]
Thank you so much for writing this issue and working on a PR @limhenry! Seems like the PR needs just a bit of final touches before we can fix this issue. Would you please check it out? Thanks!

**Comment by vanessayuenn** [comment]
Looks like this is fixed in the latest 8.0-prerelease (https://github.com/storybookjs/storybook/pull/23750#issuecomment-1912772285). Closing.
